src/infi/systray/traybar.py

```diff
@@ -161,15 +191,19 @@ def _restart(self, hwnd, msg, wparam, lparam):
         self._refresh_icon()
 
     def _destroy(self, hwnd, msg, wparam, lparam):
+        self._terminate()
+
+    def _terminate(self):
+        # TODO * release self._menu with DestroyMenu and reset the member
+        #      * release self._hicon with DestoryIcon and reset the member
+        #      * release loaded menu icons (loaded in _load_menu_icon) with DeleteObject
+        #        (we don't keep those objects anywhere now)
         if self._on_quit:
             self._on_quit(self)
         nid = NotifyData(self._hwnd, 0)
         Shell_NotifyIcon(NIM_DELETE, ctypes.byref(nid))
         PostQuitMessage(0)  # Terminate the app.
-        # TODO * release self._menu with DestroyMenu and reset the memeber
-        #      * release self._hicon with DestoryIcon and reset the member
-        #      * release loaded menu icons (loaded in _load_menu_icon) with DeleteObject
-        #        (we don't keep those objects anywhere now)
+        DestroyWindow(self._hwnd)

```

Comment:
I just realized that this change (and the one below, which changes DestroyWindow to self._terminate) is inaccurate.. I didn't catch it earlier - sorry.
We don't need to call DestroyWindow in response to WM_DESTROY because this message is sent only after DestroyWindow is already called. In the line below, SysTrayIcon.QUIT *should* call DestroyWindow, which will post a WM_DESTROY message and cause this code to be called... So the code was good before this change.